cmd/internal/kube/daemonset/daemonset.go

```diff
@@ -100,6 +181,17 @@ func (d *Daemonset) Run() error {
 	return fmt.Errorf("not implemented")
 }
 
+func (d *Daemonset) getPodArgsFromMap(podName string) (*PodArgs, error) {
```

Comment:
You've been using the pod name as a key.  But the pod name could be re-used in a relatively short amount of time. Would it be better to use the pod's UUID?